src/app/jscaip/tests/Ordinal.spec.ts

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable max-lines-per-function */
+import { MGPFallible, MGPValidation } from '@everyboard/lib';
+import { Coord } from '../Coord';
+import { DirectionFactory, DirectionFailure } from '../Direction';
+import { Ordinal } from '../Ordinal';
+
+describe('Ordinal', () => {
+
+    it('should have 8 directions', () => {
+        expect(Ordinal.ORDINALS.length).toBe(8);
+    });
+    describe('factory', () => {
+        describe('of', () => {
+            it('should construct a direction', () => {
+                expect(Ordinal.factory.from(1, 0).get()).toBe(Ordinal.RIGHT);
+                expect(Ordinal.factory.from(-1, 0).get()).toBe(Ordinal.LEFT);
+                expect(Ordinal.factory.from(0, 1).get()).toBe(Ordinal.DOWN);
+                expect(Ordinal.factory.from(0, -1).get()).toBe(Ordinal.UP);
+                expect(Ordinal.factory.from(1, -1).get()).toBe(Ordinal.UP_RIGHT);
+                expect(Ordinal.factory.from(-1, -1).get()).toBe(Ordinal.UP_LEFT);
+                expect(Ordinal.factory.from(-1, 1).get()).toBe(Ordinal.DOWN_LEFT);
+                expect(Ordinal.factory.from(1, 1).get()).toBe(Ordinal.DOWN_RIGHT);
+            });
+            it('should not construct an invalid direction', () => {
+                expect(Ordinal.factory.from(2, 1))
+                    .toEqual(MGPFallible.failure('Invalid x and y in direction construction'));
```

Comment:
j'ai pas trouvé où il est créé ce message, mais il est pas correct : 1 est valide pour y, hors le message dit que x ET y sont invalides ! C'est x ET/OU y :D

Reply:
Invalid (x and y) != Invalid x and Invalid y indeed
MAIS WHATEVER :D